docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,514 @@
+should not be shared with the Kubernetes API server.
+
+When Gardener API server is using own issuer and signing keys, the service
+account token authenticator of the Kubernetes API server will not reject the
+workload identity JWTs because:
+
+- the issuer of the tokens is not accepted
+- the tokens are not signed by trusted key
+- workload identity JWTs are not referring to any kubernetes service account
+- Gardener API will not serve the purpose of authentication or authorization
+  webhook, it will also not implement any authentication or authorization based
+  on the workload identity JWTs, it will just generate and sign them.
+
+Gardener API server will use the global configurations, `WorkloadIdentity` and
+`TokenRequest` specifications to issue JSON Web Tokens. Later, if a use case is
+identified, it could feature custom claims in the `gardener.cloud` claim
+namespace that contain additional information about the context of use of the
+token, e.g. metadata about the shoot, seed, project, landscape, etc.
+
+A sample payload of a token will look like:
+
+```json
+{
+    "aud": [
+        "service-foo-provider-bar"
+    ],
+    "exp": 1707315742,
+    "iat": 1707312142,
+    "nbf": 1707312142,
+    "iss": "https://workload-identity.gardener-local.gardener.cloud",
+    "sub": "gardener.cloud:workloadidentity:<workloadidentity-namespace>:<workloadidentity-name>:<workloadidentity-uid>",
+    "gardener.cloud": {
+        "workloadidentity": {
+            "name": "<workloadidentity-name>",
+            "namespace": "<workloadidentity-namespace>",
+            "uid": "<workloadidentity-uid>",
+        },
+        "shoot": {
+            "name": "<shoot-name>",
+            "namespace": "<shoot-namespace>",
+            "uid": "<shoot-uid>",
+        },
+        "project": {
+            "name": "<project-name>",
+            "uid": "<project-uid>",
+        },
+        "seed": {
+            "name": "<seed-name>",
+            "uid": "<seed-uid>",
+        },
+        "landscape": {
+            "id": "<gardener-landscape-id>",
+        },
+    }
+}
+```
+
+### Distribution of Workload Identity Tokens
+
+Gardenlet will request tokens as per the global configurations and renew them
+regularly. It will be responsible to provide information for the specific usage
+of the token, e.g. shoot name, namespace and UID, via the `TokenRequest` API.
+Only the `gardenlet` will be authorized to create and refresh workload identity
+tokens. Seed Authorizer will be extended to allow gardenlets to request workload
+identity tokens only for `WorkloadIdentity` that they are responsible for.
+
+As the tokens will usually have lifetime shorter than the period between two
+reconciliations, it is essential that the token creation and management are
+decoupled from the current control loops of gardenlet and implemented by a
+dedicated controller, also running in the gardenlet controller manager. Forced
```

Comment:
Can you go a bit into detail? What will be the main purpose and tasks of this new controller? What will it do?